### PR TITLE
[Merged by Bors] - feat: base change for Lie modules

### PR DESCRIPTION
**Pull request by ocfnash**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/BaseChange.lean
Comment (eric-wieser) [inline Mathlib/Algebra/Lie/BaseChange.lean:51]: I worry slightly that there's potential for an instance diamond here (generally instances of the form `[Foo A B] : Foo (Structure A) (Structure B)` are risky if there's any chance that `B = Structure A`), but I'm happy to let that be a problem for future Oliver if it ever arises.

**Review by eric-wieser — APPROVED** [review]
It's almost as if you intended this generalization from the beginning, given how easy the fixes were!

bors d+

(though feel free to wait for a review from @jcommelin)

File: Mathlib/Algebra/Lie/BaseChange.lean
Comment (ocfnash) [reply Mathlib/Algebra/Lie/BaseChange.lean:53]: I think we're safe but in any case I agree.

**Comment by ocfnash** [comment]
> It's almost as if you intended this generalization from the beginning, given how easy the fixes were!

I had not intended it and I was surprised myself!

bors merge

File: Mathlib/Algebra/Lie/BaseChange.lean
Comment (eric-wieser) [reply Mathlib/Algebra/Lie/BaseChange.lean:42]: bors r-

It looks like you didn't touch this docstring? Given the 👍 I assume that wasn't deliberate? If it was, feel free to re-merge.

**Comment by ocfnash** [comment]
This seems to have been cancelled by a bot; I'm guessing this was because I sent it to bors while it was still building (bors was empty and I had just updated a doc string).

I will thus try again.

bors merge

**Comment by eric-wieser** [comment]
@ocfnash: I cancelled it, see https://github.com/leanprover-community/mathlib4/pull/8546#discussion_r1400639031

File: Mathlib/Algebra/Lie/BaseChange.lean
Comment (ocfnash) [reply Mathlib/Algebra/Lie/BaseChange.lean:42]: Argh, apologies I missed this and confused myself into thinking your original remark referred to the module doc string.

Unfortunately this has now landed in master. Maybe the best thing to do is for me to include an update to this doc string as part of my next PR in a day or so.

**Pull request closed without merging**
